Templates/templates/CSharp/Requests/EntityCollectionRequest.cs.tt

```diff
@@ -50,18 +50,18 @@ namespace <#=@namespace#>
             {
                 if (response.AdditionalData != null)
                 {
-                    object nextPageLink;
-                    response.AdditionalData.TryGetValue("@odata.nextLink", out nextPageLink);
-
-                    var nextPageLinkString = nextPageLink as string;
-
-                    if (!string.IsNullOrEmpty(nextPageLinkString))
+                    if(response.AdditionalData.TryGetValue("@odata.nextLink", out var nextPageLink))
                     {
-                        response.Value.InitializeNextPageRequest(
-                            this.Client,
-                            nextPageLinkString);
+                        // Ensure it is a non empty JsonElement string
+                        if (nextPageLink is System.Text.Json.JsonElement element
```

Comment:
is there a way we could push all the pattern matching to a single place in the "IntializeNextPageRequest" method instead of duplicating the logic?

Reply:
I think what would happen is that instead of only giving the nextLink string to the "InitializeNextPageRequest" we would need to pass the entire response/AddtionalData for it to extract the nextLink string and create the NextPage Request object.

I think we could probably try to do it in the "CollectionResponse" object by adding a method in it that would result to something like this? (This would mean that EntityRequest.cs.tt template would remain unachanged since that template works on collection properties with customized @odata.nextlink properties)
```cs 
                var nextPageString = response.GetNextPageString(); //New method added where pattern matching happens
                if (!string.IsNullOrEmpty(nextPageString))
                {
                    response.Value.InitializeNextPageRequest(
                               this.Client,
                               nextPageString);
                }
                // Copy the additional data collection to the page itself so that information is not lost
                response.Value.AdditionalData = response.AdditionalData;
                return response.Value;
```

Reply:
Why not move the `InitializeNextPageRequest` method all together into the response base class instead? this way you'll have access to everything you need to initialize the next page request and you can pass in the client.

Reply:
@baywet Please let me know what you think of the latest update.